language/vm/src/deserializer.rs

```diff
@@ -1162,6 +1175,24 @@ fn load_struct_definition_indices(
     Ok(indices)
 }
 
+/// Builds the `FriendDeclaration` table.
+fn load_friend_decls(
+    binary: &VersionedBinary,
+    table: &Table,
+    friend_decls: &mut Vec<FriendDeclaration>,
+) -> BinaryLoaderResult<()> {
+    let start = table.offset as usize;
+    let end = start
+        .checked_add(table.count as usize)
+        .expect("Unexpected overflow as the error should be detected early by `check_tables`");
+    let mut cursor = binary.new_cursor(start, end);
+    while cursor.position() < u64::from(table.count) {
```

Comment:
I imagine Rust will CSE this, but maybe  move `u64::from(table.count)` to a local binding? Ignore this if none of the other `load_<table>` code does this

Reply:
Other `load_*` seem to be doing this as well, like `load_identifiers` and `load_constant_pool`. So I guess this should be OK.

Reply:
I imagine Rust is smart enough :) Just something to consider